plugins/system/cpu.go

```diff
@@ -4,10 +4,22 @@ import (
 	"fmt"
 
 	"github.com/influxdb/telegraf/plugins"
+	"github.com/influxdb/telegraf/plugins/system/ps/cpu"
 )
 
 type CPUStats struct {
-	ps PS
+	ps        PS
+	lastStats []cpu.CPUTimesStat
```

Comment:
@jpalay I think that this actually needs to be a map, maybe of cts.CPU -> cpu.CPUTimesStat. From what I can tell, your logic is going to assume that there is just one CPU, each subsequent CPU will then overwrite the `lastStats` value, and eventually only providing the APU stats for cpu0.